langchain/llms/openai.py

```diff
@@ -559,24 +559,79 @@ def _invocation_params(self) -> Dict[str, Any]:
         return {**{"model": self.model_name}, **super()._invocation_params}
 
 
-class AzureOpenAI(BaseOpenAI):
-    """Wrapper around Azure-specific OpenAI large language models.
+class AzureOpenAIMixin(BaseModel):
+    """Azure OpenAI mixin."""
 
-    To use, you should have the ``openai`` python package installed, and the
-    environment variable ``OPENAI_API_KEY`` set with your API key.
-
-    Any parameters that are valid to be passed to the openai.create call can be passed
-    in, even if not explicitly saved on this class.
-
-    Example:
-        .. code-block:: python
+    deployment_name: str = ""
+    openai_api_type: str = "azure"
+    openai_api_base: str = ""
+    openai_api_version: str = ""
+    openai_api_key: str = ""
+    openai_organization: str = ""
+    client_type: str = ""
 
-            from langchain.llms import AzureOpenAI
-            openai = AzureOpenAI(model_name="text-davinci-003")
-    """
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that api key and python package exists in environment."""
+        openai_api_key = get_from_dict_or_env(
+            values,
+            "openai_api_key",
+            "OPENAI_API_KEY",
+        )
+        openai_api_base = get_from_dict_or_env(
+            values,
+            "openai_api_base",
+            "OPENAI_API_BASE",
+        )
+        openai_api_version = get_from_dict_or_env(
+            values,
+            "openai_api_version",
+            "OPENAI_API_VERSION",
+        )
+        openai_api_type = get_from_dict_or_env(
+            values,
+            "openai_api_type",
+            "OPENAI_API_TYPE",
+        )
+        openai_organization = get_from_dict_or_env(
+            values,
+            "openai_organization",
+            "OPENAI_ORGANIZATION",
+            default="",
+        )
+        try:
+            import openai
 
-    deployment_name: str = ""
-    """Deployment name to use."""
+            openai.api_type = openai_api_type
+            openai.api_base = openai_api_base
+            openai.api_version = openai_api_version
+            openai.api_key = openai_api_key
+            if openai_organization:
+                openai.organization = openai_organization
+        except ImportError:
+            raise ValueError(
+                "Could not import openai python package. "
+                "Please install it with `pip install openai`."
+            )
+        try:
```

Comment:
imo we should just do this in each child class - we can do all the other validation here (cause its general) but it seems like it more work to do this switching here than to leave in actual class